api/lib/stores/block-store.js

```diff
@@ -91,7 +91,8 @@ class BlockStore {
     }
 
     // when the data is available signal readyness so the resolver stop blocking the requests
-    if (this.validators) {
+    // we assume existance of validators in most queries so we wait for the validators to be there
+    if (Object.keys(this.validators).length > 0) {
```

Comment:
fixes the store being ready too quick before the validators are actually there

Reply:
nice